### PR TITLE
Prevent RoverActivity from leaking

### LINKED ISSUE #427: RoverActivity is leaked when a Poll is present.

### PR DESCRIPTION
**Pull request by samwoodall**
## Description 
During testing I discovered that the `RoverActivity` was being leaked, this is mainly due to lack of unsubscription on view detach events. I ensured that the relevant subscribers unsubscribe correctly and that the timer is always cancelled on view detach events.

I also needed to ensure that the relevant subscribers resubscribe upon view attach events.

Resolves #427 .

### CODE REVIEW

**Comment by orospakr** [comment]
Here are my thoughts from last night as to what the proximal cause here might be.

Initial read of the issue is that .androidLifecycleDipose() was left off from the subscriber in the image poll view.  However, the problem turns out to go a bit deeper than that.

Here is a trace of the events that can lead us into an emergent bad state:

1. Poll block view is Attached by the Recycler View;
2. View Model is Bound;
3. Recycler View detaches the poll block when it has scrolled away -> before .androidLifecycleDipose was omitted, causing it just to leak. But, adding it in and continuing on results in...

... two possibilities that could now occur:

* View Re-Attached after being fully recycled. This means it gets rebound through the ViewHolder, and thus bound to a new View Model, and things continue normally. OR;
* View Re-Attached by RecyclerView, without re-binding through the ViewHolder because Recycler never bound it to a different ViewHolder.  This is the deranged state.

In the second case, scrolling back to a poll can sometimes result in a "dead" poll view that is no longer responsive to updates or vote attempts, because the subscriber is unbound (but the view otherwise remains bound to the poll as it should be).

This turns out to apparently be standard RecyclerView behaviour.

In short, the ViewModelBinding object & androidLifecycleDispose() used together are making the bad assumption that once a View is Detached it is should be considered unbound.  It turns out this is not the case.

My suggestion is that we teach ViewModelBinding to handle re-subscription as needed. (basically taking what @samwoodall has done here and factoring it into ViewModelBinding such that this fix is available for all ViewModel subscribers across the SDK, not just polls).

File: sdk/src/main/kotlin/io/rover/sdk/ui/blocks/image/ViewImage.kt [hunk 1]
```diff
@@ -21,7 +21,7 @@ internal class ViewImage(
         imageView.scaleType = ImageView.ScaleType.FIT_XY
     }
 
-    override var viewModelBinding: MeasuredBindableView.Binding<ImageBlockViewModelInterface>? by ViewModelBinding(imageView) { binding, subscriptionCallback ->
+    override var viewModelBinding: MeasuredBindableView.Binding<ImageBlockViewModelInterface>? by ViewModelBinding { binding, subscriptionCallback ->
```
Comment (orospakr) [inline sdk/src/main/kotlin/io/rover/sdk/ui/blocks/image/ViewImage.kt]: I still feel like this issue can potentially occur with image blocks and background images. But if the countermeasure is too costly, then I understand leaving it off for now.

File: sdk/src/main/kotlin/io/rover/sdk/ui/blocks/poll/image/ImagePollViewModel.kt [hunk 2]
```diff
@@ -89,7 +92,7 @@ internal class ImagePollViewModel(
     override val multiImageUpdates = PublishSubject<Map<String, ImagePollViewModelInterface.ImageUpdate>>()
 
     private fun multiImageUpdate() {
-        measurementsSubject
+        measurementsSubject.first()
```
Comment (orospakr) [inline sdk/src/main/kotlin/io/rover/sdk/ui/blocks/poll/image/ImagePollViewModel.kt]: perhaps .distinct() would be better here rather than .first(), because multiple emissions are definitely possible and the most recent one should be trusted.

File: sdk/src/main/kotlin/io/rover/sdk/ui/blocks/poll/image/ViewImagePoll.kt [hunk 3]
```diff
@@ -77,7 +79,7 @@ internal class ViewImagePoll(override val view: LinearLayout) :
             bindQuestion(viewModel.imagePoll)
             setupOptionViews(viewModel, imageLength)
 
-            viewModel.multiImageUpdates.subscribe(
+            viewModel.multiImageUpdates.first().subscribe(
```
Comment (orospakr) [inline sdk/src/main/kotlin/io/rover/sdk/ui/blocks/poll/image/ViewImagePoll.kt]: if you follow through on above comment, you would want to undo this bit here for sure :)

**Review by orospakr — CHANGES_REQUESTED** [review]

**Review by orospakr — APPROVED** [review]
This solves it for now! (modulo testing)

We can get the .1 release out, and then have the time and space to lick this one properly.
